libs/cli/src/main/java/org/elasticsearch/cli/Terminal.java

```diff
@@ -216,5 +245,58 @@ public String readText(String text) {
         public char[] readSecret(String text) {
             return readText(text).toCharArray();
         }
+
+        @Override
+        public char[] readSecret(String text, int maxLength) {
+            getErrorWriter().println(text);
+            return readLineToCharArray(getReader(), maxLength);
+        }
+
+        /**
+         * This method has to keep track of a little bit of state in order to account for
```

Comment:
Since we need to copy the line to a new array anyways (that is appropriately sized), could we simplify this method to copy until `\n` is seen, then check if we want 1 or 2 less characters depending if the second to last char is `\r`?

Reply:
Well now I feel foolish for overengineering. For some silly reason I was thinking some systems might just have an `\r` with no `\n`, but I don't think we'll be trying to run on a Commodore 64 or Mac OS 9 system any time soon. I'll definitely implement your suggestion here.